tasks/managed/sign-index-image/sign-index-image.yaml

```diff
@@ -85,33 +94,38 @@ spec:
 
-        # Translate direct quay.io reference to public facing registry reference
-        # quay.io/redhat/product----repo -> registry.redhat.io/product/repo
-        reference_image=$(translate-delivery-repo "$reference_image" | jq -r '.[] | select(.repo=="redhat.io") | .url')
+        component_count=$(jq '.components | length' "$RESULTS_FILE")
+        for ((i = 0; i < component_count; i++)); do
+          reference_image=$(jq -r ".components[$i].target_index" "$RESULTS_FILE")
+
+          # Translate direct quay.io reference to public facing registry reference
+          # quay.io/redhat/product----repo -> registry.redhat.io/product/repo
+          reference_image=$(translate-delivery-repo "$reference_image" \
+            | jq -r '.[] | select(.repo=="redhat.io") | .url')
 
-        # get all digests from manifest list
-        for manifest_digest in $(params.manifestListDigests)
-        do
+          digest_count=$(jq ".components[$i].imageDigests | length" "$RESULTS_FILE")
+          for ((j = 0; j < digest_count; j++)); do
+            manifest_digest=$(jq -r ".components[$i].imageDigests[$j]" "$RESULTS_FILE")
 
-          echo "Creating ${requestType} to sign image:"
-          echo "- reference=${reference_image}"
-          echo "- manifest_digest=${manifest_digest}"
-          echo "- requester=$(params.requester)"
+            echo "Creating ${requestType} to sign image:"
+            echo "- reference=${reference_image}"
+            echo "- manifest_digest=${manifest_digest}"
+            echo "- requester=$(params.requester)"
 
-          ${requestType} --pipeline "${request}" \
-            -p pipeline_image="${pipeline_image}" \
-            -p reference="${reference_image}" \
-            -p manifest_digest="${manifest_digest}" \
-            -p requester="$(params.requester)" \
-            -p config_map_name="${config_map_name}" \
-            -p taskGitUrl="$(params.taskGitUrl)" \
-            -p taskGitRevision="$(params.taskGitRevision)" \
-            -l ${TASK_LABEL}="${TASK_ID}" \
-            -l ${PIPELINERUN_LABEL}="$(params.pipelineRunUid)" \
-            -t "$(params.requestTimeout)" --pipeline-timeout "0h30m0s" --task-timeout "0h25m0s" \
-            "${EXTRA_ARGS[@]}" -s true
+            ${requestType} --pipeline "${request}" \
+              -p pipeline_image="${pipeline_image}" \
+              -p reference="${reference_image}" \
+              -p manifest_digest="${manifest_digest}" \
+              -p requester="$(params.requester)" \
+              -p config_map_name="${config_map_name}" \
+              -p taskGitUrl="$(params.taskGitUrl)" \
+              -p taskGitRevision="$(params.taskGitRevision)" \
+              -l ${TASK_LABEL}="${TASK_ID}" \
+              -l ${PIPELINERUN_LABEL}="$(params.pipelineRunUid)" \
+              -t "$(params.requestTimeout)" --pipeline-timeout "0h30m0s" --task-timeout "0h25m0s" \
+              "${EXTRA_ARGS[@]}" -s true
```

Comment:
any reason why not to run these in parallel?

Reply:
I thought we discussed it and agreed that this is good enough for now and parallelism is out of scope for this. But maybe that was more relevant for the add-contribution one and there it could be problematic because iib is involved and maybe you need those builds to be sequential? Anyway, yeah, it makes sense to sign in parallel. I will do it. Changing to draft for now.